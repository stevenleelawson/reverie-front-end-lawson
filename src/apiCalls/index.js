export const fetchRobots = async () => {
  try {
    const response = await fetch('http://localhost:3000/api/v1/robots');
    const robots = await response.json();
    // const robots = this.setState({ robots: data })
    console.log('robits', robots)
    return robots
  } catch (error) {
    throw new Error(error)
  }
}

export const postRobot = async (robot) => {
  console.log('postRobot', robot)
  try {
    const response = await fetch('http://localhost:3000/api/v1/robots', {
      method: 'POST',
      body: JSON.stringify(robot),
      headers: {
        'Content-Type': 'application/json'
      }
    })
    return response;
  } catch (error) {
    throw new Error('Unable to add robot' + error)
  }
};

export const deleteRobot = async (id) => {
  try {
    return await fetch(
      `http://localhost:3000/api/v1/robots/${id}`,
      {method: 'DELETE'}
    );
  } catch (error) {
    throw new Error('Unable to delete robot' + error)
  }
}

export const putRobot = async (id, robotChanges) => {
  try {
    return await fetch(
      `http://localhost:3000/api/v1/robots/${id}`,
      {
        method: 'PUT',
        body: JSON.stringify(robotChanges),
        headers: {
          'Content-Type': 'application/json'
        }
      }
    );
  } catch (error) {
    throw new Error('Unable to update robot' + error)
  }
}
