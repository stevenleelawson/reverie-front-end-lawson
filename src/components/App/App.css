.App {
  text-align: center;
}

.App-header {
  background-image: url('./bender_banner.jpg');
  background-size: cover;
  background-repeat: no-repeat;
  height: 200px;
  padding: 20px;
  display: flex;
  align-items: center;
  justify-content: center;
}

.App-title {
  font-size: 3rem;
}

.App-intro {
  font-size: large;
}

@keyframes App-logo-spin {
  from { transform: rotate(0deg); }
  to { transform: rotate(360deg); }
}
