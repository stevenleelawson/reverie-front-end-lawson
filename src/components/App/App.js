import React, { Component } from 'react';
import './App.css';
import PostForm from '../PostForm';
import { RobotContainer } from '../RobotContainer';
import * as api from '../../apiCalls/';


class App extends Component {
  constructor() {
    super();
    this.state = {
      robots: null
    }
  }

  componentDidMount = async () => {
    const robots = await api.fetchRobots();
    this.setState({ robots })
  }

  addRobot = async (robot) => {
    const postRobot = await api.postRobot(robot)
  }

  postRobot = async (robot) => {
    console.log('postRobot', robot)
    try {
      const response = await fetch('http://localhost:3000/api/v1/robots', {
        method: 'POST',
        body: JSON.stringify(robot),
        headers: {
          'content-type': 'application/json'
        }
      })
      const data = await response.json();
      return data;
    } catch (error) {
      throw new Error('Unable to add robot' + error)
    }
  };

  render() {
    return (
      <div className="App">
        <header className="App-header">
          <h1 className="App-title">Lawson's Amazing Robits</h1>
        </header>
        <PostForm postRobot={this.postRobot}/>
        {
          this.state.robots &&
          <RobotContainer data={this.state.robots}/>
        }
      </div>
    );
  }
}

export default App;
