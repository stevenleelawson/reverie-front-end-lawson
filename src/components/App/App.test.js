import React from 'react';
import ReactDOM from 'react-dom';
import App from './App';
import { shallow } from 'enzyme';
import * as api from '../../apiCalls';

jest.mock('../../apiCalls');

describe('App', () => {
  let wrapper;
  let mockRobot;
  const mockAddRobot = jest.fn();

  beforeEach(() => {
    wrapper = shallow(<App />, { disableLifecycleMethods: true});
    mockRobot = {
      date_added: "10/27/2022",
      first_active: "05/18/18",
      current_name: "Lester",
      height: "44 in",
      weight: "500 lbs",
    };

  })
  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should be able to add a robot', () => {
    wrapper.instance().addRobot(mockRobot);
    expect(api.postRobot).toHaveBeenCalled();
  });

  it('should be able to add a robot and update the page', () => {
    wrapper.instance().addRobot(mockRobot);
    expect(api.fetchRobots).toHaveBeenCalled();
  });

  it('should be able to remove a robot', () => {
    wrapper.instance().removeRobot();
    expect(api.deleteRobot).toHaveBeenCalled();
  })
});
