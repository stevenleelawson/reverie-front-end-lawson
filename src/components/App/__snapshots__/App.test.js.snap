// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should match the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <App />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <header
          className="App-header"
>
          <h1
                    className="App-title"
          >
                    Lawson's Amazing Robits
          </h1>
</header>,
        <PostForm
          addRobot={[Function]}
/>,
        null,
      ],
      "className": "App",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <h1
            className="App-title"
>
            Lawson's Amazing Robits
</h1>,
          "className": "App-header",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Lawson's Amazing Robits",
            "className": "App-title",
          },
          "ref": null,
          "rendered": "Lawson's Amazing Robits",
          "type": "h1",
        },
        "type": "header",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "addRobot": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      null,
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <header
            className="App-header"
>
            <h1
                        className="App-title"
            >
                        Lawson's Amazing Robits
            </h1>
</header>,
          <PostForm
            addRobot={[Function]}
/>,
          null,
        ],
        "className": "App",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <h1
              className="App-title"
>
              Lawson's Amazing Robits
</h1>,
            "className": "App-header",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Lawson's Amazing Robits",
              "className": "App-title",
            },
            "ref": null,
            "rendered": "Lawson's Amazing Robits",
            "type": "h1",
          },
          "type": "header",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "addRobot": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        null,
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
    "disableLifecycleMethods": true,
  },
}
`;
